retrieval/discovery/kubernetes/discovery.go

```diff
@@ -292,18 +297,31 @@ func (kd *Discovery) updateNodesTargetGroup() *config.TargetGroup {
 
 	// Now let's loop through the nodes & add them to the target group with appropriate labels.
 	for nodeName, node := range kd.nodes {
-		nodeAddress, err := nodeHostIP(node)
+		defaultNodeAddress, nodeAddressMap, err := nodeAddresses(node)
 		if err != nil {
 			log.Debugf("Skipping node %s: %s", node.Name, err)
 			continue
 		}
 
-		address := fmt.Sprintf("%s:%d", nodeAddress.String(), kd.Conf.KubeletPort)
+		kubeletPort := int(node.Status.DaemonEndpoints.KubeletEndpoint.Port)
```

Comment:
Can we expect this to usually be present? If so we can remove the config option.